conans/client/loader.py

```diff
@@ -76,6 +76,7 @@ def load_conan(self, conan_file_path, consumer=False):
             result = self._create_check_conan(loaded, consumer)
             if consumer:
                 result.options.initialize_upstream(self._options)
+	    type(result).conanfile_directory = property(lambda self, _path=conan_file_path: os.path.dirname(_path))
```

Comment:
Yes, I'd say it'd be better to handle this in a separate pull request. In particular, I think that the conanfile_directory can be made an explicit property in the base ConanFile class. If the path is dynamically assigned as it seems the case, I'd assign it here (and comment in the ConanFile class property that the value is dynamic). I like conanfile user-related things to be at least defined in the base ConanFile class for reference.